src/Workspaces/Core/Portable/Utilities/Operations/IOperationContextFactory.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using Microsoft.CodeAnalysis.Host;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Utilities
+{
+    /// <summary>
+    /// Factory to create <see cref="IOperationContext"/>s.
+    /// </summary>
+    internal interface IOperationContextFactory : IWorkspaceService
+    {
+        IOperationContext CreateOperationContext(string title, string description, bool allowCancellation, bool showProgress);
+    }
+
+    [ExportWorkspaceService(typeof(IOperationContextFactory)), Shared]
+    internal class DefaultOperationContextFactory : IOperationContextFactory
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public DefaultOperationContextFactory()
+        {
+        }
+
+        public IOperationContext CreateOperationContext(string title, string description, bool allowCancellation, bool showProgress)
+            => DefaultOperationContext.Instance;
+
+        private class DefaultOperationContext : IOperationContext
+        {
+            public static readonly IOperationContext Instance = new DefaultOperationContext();
```

Comment:
>Instance = new DefaultOperationContext(); [](start = 53, length = 41)

Seems that the context is supposed to be mutable. It's odd to have a singleton of a mutable object.

Reply:
Fair enough.  I'll adjust.